### PR TITLE
Fixed a bug that caused unexpected MigrationNeededException when a field was added to synced Realm

### PR DESCRIPTION
**Pull request by zaki50**
In sync mode, `validateTable()` must get all fields in the tabla.

@realm/java

### CODE REVIEW

File: realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java [hunk 1]
```diff
@@ -375,7 +375,7 @@ private void injectObjectContext() {
             }
             final Row row = proxyState.getRow$realm();
             if (value == null) {
-                // Table#nullifyLink() does not support default value. Just use Row.
+                // Table#nullifyLink() does not support default value. Just using Row.
```
Comment (dalinaum) [inline realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java]: Is `Just use` correct?

File: realm/realm-annotations-processor/src/test/resources/io/realm/NullTypesRealmProxy.java [hunk 2]
```diff
@@ -980,7 +980,7 @@ private void injectObjectContext() {
             }
             final Row row = proxyState.getRow$realm();
             if (value == null) {
-                // Table#nullifyLink() does not support default value. Just use Row.
+                // Table#nullifyLink() does not support default value. Just using Row.
```
Comment (dalinaum) [inline realm/realm-annotations-processor/src/test/resources/io/realm/NullTypesRealmProxy.java]: Same above.

**Comment by kneth** [comment]
LGTM

File: realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java (hunk 1 above)
Comment (zaki50) [inline realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java]: This comment is generated by the annotation processor.
I think that some changes in generator code made by other PR was not applied to those test resouces.

**Pull request merged**